### PR TITLE
Set and get system properties

### PR DESCRIPTION
**Pull request by AymericDu**

### CODE REVIEW

File: src/main/java/io/openio/sds/AdvancedClient.java [hunk 1]
```diff
@@ -193,6 +254,23 @@ public void setContainerProperties(OioUrl url, Map<String, String> props,
...
+     * @param reqCtx
+     *            common parameters to all requests
+     * @return the user properties and system properties found on the container
+     * @throws ContainerNotFoundException
+     *             if the specified container doesn't exist
+     * @throws OioSystemException
+     *             if any error occurs during request execution
+     */
+    public Map<String, Map<String, String>> getAllContainerProperties(OioUrl url,
+            RequestContext reqCtx) throws OioException;
```
Comment (fvennetier) [inline src/main/java/io/openio/sds/AdvancedClient.java:272]: Returning a map which will ever contain only 2 values seems overkill. I suggest returning an array of `Map<string, String>` (or maybe a list?).

File: src/main/java/io/openio/sds/AdvancedClient.java [hunk 2]
```diff
@@ -82,6 +80,33 @@ public ContainerInfo createContainer(OioUrl url,
...
+     * to specify the name of the account to use and the name of the future
+     * container.
+     * <p>
+     * The container is available when the returned future is completed.
+     *
+     * @param url
+     *            the URL of the container
+     * @param properties
+     *            the user properties to set to the container.
+     * @param properties
```
Comment (fvennetier) [inline src/main/java/io/openio/sds/AdvancedClient.java]: `@param system`

File: src/main/java/io/openio/sds/AdvancedClient.java (hunk 1 above)
Comment (AymericDu) [reply src/main/java/io/openio/sds/AdvancedClient.java:272]: With `JsonUtils.jsonToMapMap` (https://github.com/open-io/oio-api-java/pull/71/files#diff-f0a51d0012e81f76617778014e7d65aeR1028), we have automatically `Map<string, Map<string, String>>`. I do not think it's useful to convert a `Map<string, Map<string, String>>` to a `List<Map<string, String>>`.

Comment (fvennetier) [reply src/main/java/io/openio/sds/AdvancedClient.java:272]: Ok.

**Review by fvennetier — APPROVED** [review]

File: src/test/java/io/openio/sds/ClientITest.java [hunk 3]
```diff
@@ -345,6 +345,51 @@ public void containerPropertiesWithClear() {
...
+            res = client.getAllContainerProperties(url);
+            assertNotNull(res);
+            assertEquals(2, res.size());
+            assertTrue(res.get("properties").isEmpty());
+            sysRes = res.get("system");
+            for (Entry<String, String> e : sys.entrySet()) {
+                assertFalse(sysRes.containsKey(e.getKey()));
+            }
+        } finally {
+            // client.deleteContainer(url);
```
Comment (murlock) [inline src/test/java/io/openio/sds/ClientITest.java]: We should delete the container

**Review by murlock — APPROVED** [review]

**Review by murlock — APPROVED** [review]

**Review by kamel-rahim — APPROVED** [review]

**Pull request merged**